src/pybind/mgr/dashboard/tests/test_module.py

```diff
@@ -0,0 +1,30 @@
+import errno
+import unittest
+
+from ..tests import CLICommandTestMixin, CmdException
+
+
+class DashboardCommandsTest(unittest.TestCase, CLICommandTestMixin):
+
+    CMD_OK = 'Option GRAFANA_API_URL updated'
+    CMD_ERR = "invalid value. '{}' is not a valid URL"
+
+    def test_grafana_set_url_invalid_http_protocol(self):
+        with self.assertRaises(CmdException) as ctx:
+            self.exec_cmd('set-grafana-api-url', value='foo://localhost:3000')
+        self.assertEqual(ctx.exception.retcode, -errno.EINVAL)
+        self.assertEqual(str(ctx.exception), self.CMD_ERR.format('foo://localhost:3000'))
+
+    def test_grafana_set_url_invalid_port(self):
+        with self.assertRaises(CmdException) as ctx:
+            self.exec_cmd('set-grafana-api-url', value='https://localhost:foobar')
+        self.assertEqual(ctx.exception.retcode, -errno.EINVAL)
+        self.assertEqual(str(ctx.exception), self.CMD_ERR.format('https://localhost:foobar'))
```

Comment:
I think it would be good to validate something like `https://localhost:8000x` to see if port parsing is working correctly or not.